### PR TITLE
Add beacon-chain/state/v2/state_trie_block_box_test.go from hf1 branch

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

As title describes, I found this test in hf1 and could be merged into develop.

**Which issues(s) does this PR fix?**

Required as part of #8638

**Other notes for review**

Added some t.Skip pending https://github.com/ferranbt/fastssz/issues/67
\

### CODE REVIEW

File: beacon-chain/state/v2/state_trie_block_box_test.go
Comment (prestonvanloon) [inline beacon-chain/state/v2/state_trie_block_box_test.go:105]: @terencechain any idea how this can be fixed?

**Comment by prestonvanloon** [comment]
Blocked by https://github.com/ferranbt/fastssz/issues/43

**Comment by prestonvanloon** [comment]
Still blocked

**Comment by prestonvanloon** [comment]
Should be unblocked by #9542

**Comment by prestonvanloon** [comment]
Nope, still blocked / tests fail.

**Comment by terencechain** [comment]
blocked by https://github.com/ferranbt/fastssz/issues/61

**Comment by rsacannabis** [comment]
RSACannabis

**Comment by rauljordan** [comment]
Hi @prestonvanloon just checking in on next steps for this PR

**Comment by prestonvanloon** [comment]
@rauljordan i dont know. I'm not original author of this test.

@terencechain can you help us understand if this is still blocked and how we can move forward?

**Comment by terencechain** [comment]
> @rauljordan i dont know. I'm not original author of this test.
> 
> @terencechain can you help us understand if this is still blocked and how we can move forward?

the latest fssz should unblock this

**Comment by prestonvanloon** [comment]
> the latest fssz should unblock this

OK I'll sync this branch and see what happens. Thanks!!

**Comment by prestonvanloon** [comment]
Still doesn't work.

**Comment by terencechain** [comment]
> Still doesn't work.

Can confirm that it doesn't work after https://github.com/ferranbt/fastssz/pull/65. There's another bug in fastssz's state hash tree root

**Review by jaystyles — DISMISSED** [review]

**Comment by prestonvanloon** [comment]
I ended up doing a ton of refactoring to ssz_static spectests in this PR. Reverting to draft and will split the refactoring support to another PR for ease of review.

**Pull request closed without merging**
